src/src/PluginStructs/P128_data_struct.cpp

```diff
@@ -344,273 +335,273 @@ bool P128_data_struct::plugin_write(struct EventStruct *event,
+      mode = P128_modetype::SimpleClock;
 
-          # if defined(RGBW) || defined(GRBW)
+      # if defined(RGBW) || defined(GRBW)
 
-      if (parseString(string, 3) != "") {
-        if (parseString(string, 3).length() <= 6) {
+      if (!str3.isEmpty()) {
+        if (str3.length() <= 6) {
           rgb_tick_s =
-            RgbwColor(rgbStr2Num(parseString(string, 3)) >> 16, rgbStr2Num(parseString(string, 3)) >> 8,
-                      rgbStr2Num(parseString(string, 3)));
+            RgbwColor(rgbStr2Num(str3) >> 16, rgbStr2Num(str3) >> 8,
+                      rgbStr2Num(str3));
         } else {
           rgb_tick_s =
-            RgbwColor(rgbStr2Num(parseString(string, 3)) >> 24,
-                      rgbStr2Num(parseString(string, 3)) >> 16,
-                      rgbStr2Num(parseString(string, 3)) >> 8,
-                      rgbStr2Num(parseString(string, 3)));
+            RgbwColor(rgbStr2Num(str3) >> 24,
+                      rgbStr2Num(str3) >> 16,
+                      rgbStr2Num(str3) >> 8,
+                      rgbStr2Num(str3));
         }
       }
 
-      if (parseString(string, 4) != "") {
-        if (parseString(string, 4).length() <= 6) {
+      if (!str4.isEmpty()) {
+        if (str4.length() <= 6) {
           rgb_tick_b =
-            RgbwColor(rgbStr2Num(parseString(string, 4)) >> 16, rgbStr2Num(parseString(string, 4)) >> 8,
-                      rgbStr2Num(parseString(string, 4)));
+            RgbwColor(rgbStr2Num(str4) >> 16, rgbStr2Num(str4) >> 8,
+                      rgbStr2Num(str4));
         } else {
           rgb_tick_b =
-            RgbwColor(rgbStr2Num(parseString(string, 4)) >> 24,
-                      rgbStr2Num(parseString(string, 4)) >> 16,
-                      rgbStr2Num(parseString(string, 4)) >> 8,
-                      rgbStr2Num(parseString(string, 4)));
+            RgbwColor(rgbStr2Num(str4) >> 24,
+                      rgbStr2Num(str4) >> 16,
+                      rgbStr2Num(str4) >> 8,
+                      rgbStr2Num(str4));
         }
       }
 
-      if (parseString(string, 5) != "") {
-        if (parseString(string, 5).length() <= 6) {
+      if (!str5.isEmpty()) {
+        if (str5.length() <= 6) {
           rgb_h =
-            RgbwColor(rgbStr2Num(parseString(string, 5)) >> 16, rgbStr2Num(parseString(string, 5)) >> 8,
-                      rgbStr2Num(parseString(string, 5)));
+            RgbwColor(rgbStr2Num(str5) >> 16, rgbStr2Num(str5) >> 8,
+                      rgbStr2Num(str5));
         } else {
           rgb_h =
-            RgbwColor(rgbStr2Num(parseString(string, 5)) >> 24,
-                      rgbStr2Num(parseString(string, 5)) >> 16,
-                      rgbStr2Num(parseString(string, 5)) >> 8,
-                      rgbStr2Num(parseString(string, 5)));
+            RgbwColor(rgbStr2Num(str5) >> 24,
+                      rgbStr2Num(str5) >> 16,
+                      rgbStr2Num(str5) >> 8,
+                      rgbStr2Num(str5));
```

Comment:
That's a lot of calls to `rgbStr2Num(str5)` Why not store it in some const int and then process it?

Reply:
That is planned for the next round of optimizations.